cwatch.php

```diff
@@ -252,57 +182,119 @@ public function getEmailTags()
-        }
-        if (isset($row->meta->password)) {
-            $pass = $row->meta->password;
-        }
-        if (isset($package->meta->cwatch_license_type)) {
-            $product = $package->meta->cwatch_license_type;
-        }
-        if (isset($package->meta->cwatch_license_term)){
-            $term = $package->meta->cwatch_license_term;
-        }
-        if (isset($package->meta->cwatch_sandbox)){
-            $sandbox = $package->meta->cwatch_sandbox;
+    public function addService(
+        $package,
+        array $vars = null,
+        $parent_package = null,
+        $parent_service = null,
+        $status = 'pending'
+    ) {
+        Loader::loadModels($this, ['Clients', 'Services']);
+
+        $this->validateService($package, $vars);
+
+        if ($this->Input->errors()) {
+            return;
         }
+
+        // Get cWatch API
+        $api = $this->getApi();
+        $errors = [];
         if (isset($vars['use_module']) && $vars['use_module'] == 'true') {
-            Loader::loadModels($this, ['Clients']);
-            $client = $this->Clients->get($vars['client_id'], false);
-            $firstname = $client->first_name;
-            $lastname = $client->last_name;
-            $email = $client->email;
-            $country = $client->country;
-            $this->loadApi($user, $pass, $sandbox);
-            try{
-                $response = $this->api->createlicence($email, $firstname, $lastname, $country, $product, $term);
-                $json = json_decode($response->resp);
+            $response = null;
+            try {
+                // Log request data
+                $this->log('createcommand', serialize($vars), 'input', true);
+
+                // Add a customer account in cWatch
+                $response = $api->addUser(
+                    $vars['cwatch_email'],
+                    $vars['cwatch_firstname'],
+                    $vars['cwatch_lastname'],
+                    $vars['cwatch_country']
+                );
+
+                // Log response
+                $this->log('adduser', serialize($response), 'output', $response->code == 200);
+
                 if ($response->code != 200) {
-                    $this->Input->setErrors(['api' => ['internal' => $response->errorMsg]]);
+                    $errors = ['api' => ['internal' => $response->errorMsg]];
+                } else {
+                    $license_types = Configure::get('cwatch.products');
+                    // Get a count of licenses to provide for each type
+                    foreach ($license_types as $license_type => $value) {
+                        $license_types[$license_type] = isset($vars['configoptions'][$license_type])
+                            ? $vars['configoptions'][$license_type]
+                            : 0;
+                    }
+
+                    // Determine what term to add these licenses for
+                    $license_term = 'MONTH_1';
```

Comment:
Shouldn't this fail if we don't have a pricing term that matches a valid license term? If I select a 3 month pricing period/term, why should the module add it for 1 month instead?